app/main/posts/views/share/post-export.directive.js

```diff
@@ -28,54 +29,85 @@ function PostExportController(
     ConfigEndpoint,
     Notify,
     $q,
-    PostFilters
+    PostFilters,
+    _
 ) {
     $scope.loading = false;
-    $scope.exportPosts = exportPosts;
-
-    function exportPosts() {
+    this.getQuery = getQuery;
+    this.prepareExport = prepareExport;
+    this.showCSVResults = showCSVResults;
+    this.requestExport = requestExport;
+    $scope.exportPostsConfirmation = function () {
+        /**
+         * Trigger confirm notification for user.
+         * When the user accepts, get the CSV
+         */
         Notify.confirm('notify.post.export').then(function (message) {
+            prepareExport();
+        });
+    };
+    function prepareExport() {
+        loadingStatus(true);
+        var site = ConfigEndpoint.get({ id: 'site' }).$promise;
+        var query = getQuery();
+        var exportQuery = PostEndpoint.export(query);
+        requestExport(site, query, exportQuery);
+    }
+
+    function requestExport(site, query, exportQuery) {
+        $q.all([site, exportQuery]).then(function (response) {
+            showCSVResults(response, query.format);
+        }).finally(function (response) {
```

Comment:
Can we add a failure state that use Notify.errors to update the user for a fail? Since this is pretty standard everywhere we should be able to just pass the error response directly to the notify function on failure.

Reply:
ok

Reply:
@crcommons I'll also implement the feedback from Jess directly here if you don't mind .

Reply:
https://github.com/ushahidi/platform-client/pull/755

Reply:
I think you can use this guy
https://github.com/ushahidi/platform-client/blob/8f8ad88a3a92eb107f4da32de03d1104354a7ace/app/main/posts/common/post-actions.service.js#L25